src/data/int/log.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2022 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import data.nat.log
+import algebra.order.floor
+import algebra.field_power
+
+/-!
+# Integer logarithms in a field with respect to a natural base
+
+This file defines two `ℤ`-valued analogs of the logarithm of `n : R` with base `b : ℕ`:
+
+* `int.log b n`: Lower logarithm, or floor **log**. Greatest `k` such that `↑b^k ≤ n`.
+* `int.clog b n`: Upper logarithm, or **c**eil **log**. Least `k` such that `n ≤ ↑b^k`.
+
+Note that `int.log` gives the position of the left-most non-zero digit:
+```lean
+#eval (int.log 10 (0.09 : ℚ), int.log 10 (0.10 : ℚ), int.log 10 (0.11 : ℚ))
+--    (-2,                    -1,                     -1)
+#eval (int.log 10 (9 : ℚ),    int.log 10 (10 : ℚ),   int.log 10 (11 : ℚ))
+--    (0,                      1,                     1)
```

Comment:
Nitpick: Would probably be more readable with one example per line (but I recognize that the idea of putting stuff in tuples is nice!).
```suggestion
#eval int.log 10 (0.09 : ℚ) -- -2
#eval int.log 10 (0.10 : ℚ) -- -1
#eval int.log 10 (0.11 : ℚ) -- -1
#eval int.log 10 (9    : ℚ) --  0
#eval int.log 10 (10   : ℚ) --  1
#eval int.log 10 (11   : ℚ) --  1
```

Reply:
Will let someone else tiebreak

Reply:
I'm going to go with Eric's approach on this one because it groups related examples in a nicer way, although neither form parses very easily to me.